src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pylsl.py

```diff
@@ -0,0 +1,77 @@
+import os
+import platform
+import struct
+import pathlib
+from ctypes import util
+from PyInstaller.utils.hooks import get_module_file_attribute, logger
+
+"""
+Hook for pylsl, i.e., the Python wrapper around labstreaminglayer (aka LSL).
+The pylsl package loads the C++ liblsl library using CDLL. The logic for
+locating liblsl is replicated here; subsequently it is added to the binaries
+and packaged in the compiled version such that pylsl will be able to find it
+there as well.
+
+See also
+-  https://labstreaminglayer.readthedocs.io
+-  https://github.com/labstreaminglayer/pylsl
+-  https://github.com/sccn/liblsl
+"""
+
+
+def find_liblsl_library():
```

Comment:
Any reason why you can't just import and call [the original find_liblsl_library()](https://github.com/labstreaminglayer/pylsl/blob/16a4198087936386e866d7239bfde32d1fef6d6b/pylsl/pylsl.py#L1198)?

Reply:
Hmm, that is something I had not considered. That could indeed work, I will give it a try.

Reply:
Yes, that works. The generator function itself is not directly exposed, but it can be imported with some extra effort.

I also added the test as you suggested.

Reply:
I think it might actually be best to just read the  `pylsl.pylsl.libpath` attribute:

https://github.com/labstreaminglayer/pylsl/blob/16a4198087936386e866d7239bfde32d1fef6d6b/pylsl/pylsl.py#L1286-L1302

i.e.,

```python
from PyInstaller.utils.hooks import get_module_attribute

try:
    libpath = get_module_attribute('pylsl.pylsl', 'libpath')
except Exception:
    libpath = None
``` 

---

The test function (which should be added to https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py) should also look at `pylsl.pylsl.libpath` and ensure that the library is loaded from the frozen application's directory (i.e., that the path begin withs `sys._MEIPASS`). Otherwise, the test might still pass even though the library is not properly collected (because the system copy can still be found).